core/lib/src/form/from_form.rs

```diff
@@ -415,10 +415,7 @@ impl<'v, T: FromForm<'v>> VecContext<'v, T> {
 
     fn context(&mut self, name: &NameView<'v>) -> &mut T::Context {
         let this_key = name.key();
-        let keys_match = match (self.last_key, this_key) {
-            (Some(k1), Some(k2)) if k1 == k2 => true,
-            _ => false
-        };
+        let keys_match = matches!((self.last_key, this_key), (Some(k1), Some(k2)) if k1 == k2);
```

Comment:
This one is too complex, I think, for a single liner. Let's revert.

Reply:
No, please use the existing version.

Reply:
I think I've did not give this lint the proper attention because there were lots of them to fix. I suggest two solution :
- Fixing the lint like this :
```
match (self.last_key, this_key) {
    (Some(k1), Some(k2)) if k1 == k2 => (),
    _ => {
        self.shift();
        self.current = Some(T::init(self.opts));
    }
};
```
- Allowing the lint and renaming `keys_match` to make the intent clearer :
```
#[allow(clippy::match_like_matches_macro)] // Using the macro will make the match more complex.
let is_new_key = match (self.last_key, this_key) {
    (Some(k1), Some(k2)) if k1 == k2 => false,
    _ => true
};
if is_new_key {
    self.shift();
    self.current = Some(T::init(self.opts));
}
```
Which one you prefer?

Reply:
I prefer the original version.

Reply:
Ok thanks, I'll use the original code with this allow :
```
#[allow(clippy::match_like_matches_macro)] // In this context, match makes the intent clearer.
```